go-controller/pkg/ovn/egressip.go

```diff
@@ -1601,6 +1603,39 @@ func (oc *DefaultNetworkController) syncStaleSNATRules(egressIPCache map[string]
 	return nil
 }
 
+func (oc *DefaultNetworkController) syncEgressIPsStatus() error {
+	if !util.PlatformTypeIsEgressIPCloudProvider() {
+		return nil
+	}
+	egressIPs, err := oc.watchFactory.GetEgressIPs()
+	if err != nil {
+		return err
+	}
+	cloudPrivateIPConfigMap, err := oc.getCloudPrivateIPConfigMap()
+	if err != nil {
+		return err
+	}
+	for _, egressIP := range egressIPs {
+		updatedStatus := []egressipv1.EgressIPStatusItem{}
+		ipNotFound := false
+		for _, status := range egressIP.Status.Items {
+			cloudPrivateIPConfigName := ipStringToCloudPrivateIPConfigName(status.EgressIP)
+			if _, exists := cloudPrivateIPConfigMap[cloudPrivateIPConfigName]; exists {
+				updatedStatus = append(updatedStatus, status)
+			} else {
+				ipNotFound = true
+			}
+		}
+		if ipNotFound {
+			err = oc.patchReplaceEgressIPStatus(egressIP.Name, updatedStatus)
```

Comment:
let's put an info long here about this update. Actually, it may be useful to have one in line 1626 too, wdyt?